server/src/main/java/io/crate/role/TransportPrivilegesAction.java

```diff
@@ -91,61 +92,76 @@ protected void masterOperation(PrivilegesRequest request,
         }
 
         clusterService.submitStateUpdateTask("grant_privileges",
-            new AckedClusterStateUpdateTask<PrivilegesResponse>(Priority.IMMEDIATE, request, listener) {
-
-                long affectedRows = -1;
-                List<String> unknownUserNames = null;
-
-                @Override
-                public ClusterState execute(ClusterState currentState) throws Exception {
-                    Metadata currentMetadata = currentState.metadata();
-                    Metadata.Builder mdBuilder = Metadata.builder(currentMetadata);
-                    unknownUserNames = validateUserNames(currentMetadata, request.userNames());
-                    if (unknownUserNames.isEmpty()) {
-                        affectedRows = applyPrivileges(mdBuilder, request);
+                new AckedClusterStateUpdateTask<>(Priority.IMMEDIATE, request, listener) {
+
+                    long affectedRows = -1;
+                    List<String> uknownRoleNames = null;
+
+                    @Override
+                    public ClusterState execute(ClusterState currentState) throws Exception {
+                        Metadata currentMetadata = currentState.metadata();
+                        Metadata.Builder mdBuilder = Metadata.builder(currentMetadata);
+                        uknownRoleNames = validateRoleNames(currentMetadata, request);
+                        if (uknownRoleNames.isEmpty()) {
+                            affectedRows = applyPrivileges(mdBuilder, request);
+                        }
```

Comment:
In fact we never expect it to be non-empty because of analyzer checks, don't we?
If so maybe need an assertion below?

Reply:
PrivilegesAnalyzer doesn't check if the roles exist, we need to do it here, from clusterState, or you mean something else?

Reply:
ah, ok it's tested in `test_grant_revoke_roles_for_unknonwn_roles`.
Do I understand it correctly that uknownRoleNames and check inside applyPrivileges
```
if (role == null) {
            throw new IllegalArgumentException("ROLE: " + roleName + " does not exist");
}
```
kinda similar?

I'm wondering if we can drop `validateRoleNames` if we anyway check existence later.

Reply:
hm, right, I'd rather keep it here to not change the response format, but remove those "safety net" checks in RolesMetadata, do you agree?

Reply:
I saw that test is gone now


Shouldn't we still cover this?
We don't do anything if we have unknown names - it means operation `GRANT user1, user2, unkwonnUser` will silently become a no-op? Maybe we should throw an error so that it's explicitly clear for users?

It was covered before, now it's gone 
https://github.com/crate/crate/pull/15249/commits/e855160718f4ff52fbd3158df7c76e9eaa83352b#diff-8a3f67c898993c25482fd4ea63a894c0c408a4099f058fef1f9ee0483ae55883L168

let's throw and get back coverage of that scenario?

Reply:
Nevermind, we want to return 0 rows affected. Nested check was not supposed to really kick in.

Reply:
yes, sorry, I didn't explain this correctly.